agent/connect/ca/common.go

```diff
@@ -89,3 +90,15 @@ func validateSignIntermediate(csr *x509.CertificateRequest, spiffeID *connect.Sp
 	}
 	return nil
 }
+
+// AddSingleNewline this is used to fix a case where the provider do not return a new line after
+// the certificate as per the specification see GH-8178 for more context
+func AddSingleNewline(cert string) string {
```

Comment:
`Add` implies that we always add it. What do you think about calling this `EnsureTrailingNewline` ? I think that name might match the behaviour a little better.

```suggestion
// EnsureTrailingNewline adds a single newline to the string if the string does not already have a
// trailing newline. The trailing newlines is necessary to ensure that certificats can be
// concatenated. See GH-8178 for more context.
func EnsureTrailingNewline(cert string) string {
```